firebase-functions/functions/src/operations/playerOperations.ts

```diff
@@ -0,0 +1,32 @@
+﻿import { getPlayersToDeactivate } from './choiceOperations';
+import * as admin from 'firebase-admin';
+import { firestore } from 'firebase-admin';
+import { collections } from '../constants/collections';
+import { Player } from '../models/player';
+import DocumentData = firestore.DocumentData;
+import QueryDocumentSnapshot = firestore.QueryDocumentSnapshot;
+
+export async function deactivatePlayers(roomId: string, initiallyActivePlayers: QueryDocumentSnapshot<DocumentData>[]): Promise<void> {
+  const playersToDeactivate = await getPlayersToDeactivate(initiallyActivePlayers, roomId);
+  if (!playersToDeactivate.length) {
+    return;
+  }
+  await updatePlayers(playersToDeactivate, { isObserver: true });
+  await determineLastOneActive(initiallyActivePlayers, playersToDeactivate, roomId);
+}
+
+export async function resetPlayerChoices(players: QueryDocumentSnapshot<DocumentData>[]): Promise<void> {
+  await updatePlayers(players, { choice: null });
+}
+
+async function determineLastOneActive(initiallyActivePlayers: QueryDocumentSnapshot<DocumentData>[], playersToDeactivate: QueryDocumentSnapshot<DocumentData>[], roomId: string): Promise<void> {
+  const stillActivePlayers = initiallyActivePlayers.filter(player => !playersToDeactivate.includes(player));
+  if (stillActivePlayers.length === 1) {
+    await admin.firestore().collection(collections.rooms).doc(roomId).update({ lastOneActive: stillActivePlayers[0].id });
+    await updatePlayers(stillActivePlayers, { isObserver: true });
```

Comment:
I get what you mean and you are right. We should use one term to describe this concept.
I chose `isObserver` back then because boolean defaults to false, and I wanted players to be active/not observers by default. 
How about I rename the property to `isDeactivated`? A little clunky, but might serve the purpose better and still easily readoble probably?